content/docs/platform/concepts/subscribers.mdx

```diff
@@ -97,4 +100,110 @@ await novu.trigger({
 
 When triggering the workflow, you must specify all required `to` fields per the channels in the workflow. For example, if an email step is included in the workflow, then the `to.email` field must be specified. Similarly if an SMS step is included, the `to.phone` field must be specified.
 
-If you are defining workflows with code, you can use the `subscriberId` to hydrate subscriber data directly from your database or other sources during workflow execution. This is useful when you don't want to store all the subscriber data in Novu.
+If you are defining workflows with code using [@novu/framework](/framework/introduction), you can use the `subscriberId` to hydrate subscriber data directly from your database or other sources during workflow execution. This is useful when you don't want to store all the subscriber data in Novu.
+
+For example, you may have a welcome onboarding email that you trigger immediately after a user signs up, with the email only sent 1 hour after signup:
+
+```typescript
+import { workflow } from '@novu/framework';
+import { userDb } from '../db/userDb';
+
+export const welcomeEmail = workflow('welcome-email', async ({ step, subscriber }) => {
+  await step.delay('delay', () => ({ amount: 1, unit: 'hours' }));
+
+  await step.email('send-email', async (controls) => {
+    // Fetch user data from your database in real-time to
+    // ensure up-to-date information
+    const user = await userDb.findById(subscriber.subscriberId);
+
+    return {
+      subject: `Welcome, ${user.firstName}`,
+      body: 'Welcome to our platform!',
+    };
+  });
+});
+```
+
+Then, when you trigger the workflow, you can pass the `subscriberId` and the workflow will fetch the user data from your database in real-time:
+
+```typescript
+await welcomeEmail.trigger({
+  to: {
+    subscriberId: '123',
+    email: 'john.doe@domain.com',
+  },
+});
+```
+
+<Callout type="info">
+It is possible for two subscribers to have the same email address in our system as `subscriberId` is the unique identifier that distinguishes one subscriber from another.
+</Callout>
+
```

Comment:
I think it's worth adding here a link, Learn More about code based workflows for further reading.

Reply:
@scopsy 

Not sure, I understand the requirement of framework link here.

Reply:
we are explaining here about how to do something with the framework, but user might not know what framework is. So adding here some sort of a link to learn more about the framework can be helpful